persistence-jdbc/core/src/main/scala/com/lightbend/lagom/internal/persistence/jdbc/SlickProvider.scala

```diff
@@ -41,30 +42,38 @@ private[lagom] class SlickProvider(
 
   val autoCreateTables: Boolean = createTables.getBoolean("auto")
 
-  private val asyncExecConfig =
-    new AsyncExecutorConfig(system.settings.config.getConfig("jdbc-defaults.slick.async-executor"))
-
   if (dbApi != null) {
```

Comment:
I know you didn't add this, but in what context is `dbApi` `null`? Seems like it shouldn't ever be in production, but some tests pass in `null` explicitly. Makes me think we have a factoring flaw.

Reply:
You are right, the reason for that is the tests. I will check how I can remove this hack.

Reply:
@TimMoore, I think I can move the JNDI bindings to a `JndiConfigurator`. That will allow us to remove it from the `SlickProvider`.

What we need is something that guarantees that the JNDI resources get bound before we start to use them and we are piggybacking the `SlickProvider` constructor for this. 

In tests, because it's not easy to build a DbApi object, we are binding the JNDI first and then passing a null. It's smelly. 

I want to improve this, but don't want to block this PR because of it. So, I will send another PR with only this improvement. I need more time to tweak it in Guice. In Macwire is a no brain.